File: LeetCode/Medium/Longest Subarray Of 1s After Deleting One Element/Java/longestSubarray.java
class Solution {
    public int longestSubarray(int[] nums) {
        int zero = 0;
        int maxLength = 0;
        int i = 0;
        int k = 0;
        int currLen = 0;
        for (i=0; i<nums.length; i++) {
            if (nums[i] == 0) zero++;
            else currLen++;
            if (zero > 1) {
                maxLength = Math.max(maxLength, i-k-1);
                while (zero > 1) {
                    if (nums[k] == 0) zero--;
                    k++;
                }
            }
        }
        maxLength = Math.max(maxLength, i-k-1);
        return maxLength;
    }
}

File: LeetCode/Medium/Longest Subarray Of 1s After Deleting One Element/C++/longest_subarray.cc
class Solution {
public:
    int longestSubarray(vector<int>& nums) {
        int z=0, j=0;
        int max_len = 0;
        int i=0;
        for (i=0; i<nums.size(); i++) {
            if (nums[i] == 0) z++;
            if (z > 1) {
                max_len = max(max_len, i-j-1);
                while (z > 1) {
                    if (nums[j] == 0) {
                        z--;
                    }
                    j++;
                }
            }
        }
        max_len = max(max_len, i-j-1);
        return max_len;
    }
};
